eureka2-testkit/src/main/java/com/netflix/eureka2/testkit/embedded/cluster/EmbeddedWriteCluster.java

```diff
@@ -134,7 +136,18 @@ public Integer call(WriteServerAddress writeServerAddress) {
                         }
 
                         WriteServerAddress data = notification.getData();
-                        Server serverAddress = new Server(data.getHostName(), data.getReplicationPort());
+                        int port;
+                        switch (serviceType) {
+                            case Registration:
+                                port = data.getRegistrationPort();
+                                break;
+                            case Interest:
+                                port = data.getDiscoveryPort();
+                                break;
+                            default: // == Replication
```

Comment:
Can we change our code pattern to always explicitly check for all cases and defaults to some sort of breaking error? Treating default as a specific valid case can potentially lead to missed updates when an enum is added for example, and thus lead to non-deterministic behaviour.